### ISSUE TITLE
Jigsaw block support

### ISSUE THREAD

**Issue by Raycoms** [ldtteam/Structurize#492]
Jigsaw blocks are blocks to construct worldgen colonies.
They are tileEntities that hold a bunch of useful data, some of that data could also be useful to handle growing colonies etc. (Where would it connect to a road etc).

However, they are creative only blocks. Furthermore, it is possible to define a "replacement" block in the jigsaw block.

Support would need to be two fold:

a) In "complete" paste mode nothing changes
b) In the "natural paste mode" we need to return the "replacement block" or air.

**Comment by uecasm** [comment]
I do have some code kicking around that supports these, although only by allowing .blueprint files during the normal worldgen structure placement; it doesn't do anything with them when placing with the build tool in any mode.

I kind of prefer the way it is now with the worldgen colonies implemented only in .nbt though, since that explicitly does mean that these are different schematics and can have different appearance (cobwebs, decay, hand-placed "collapsed sections", lootboxes, etc).  So I don't think that it's a good idea to just add the jigsaw blocks to the regular player-placeable schematics.

Perhaps what is really needed is a way to more easily place and re-save an .nbt structure?  Although that's already possible with the vanilla structure blocks.

**Comment by Raycoms** [comment]
The idea of this is that the schematic on the schematic server can have all the data necessary. So that we don't need two locations. But when scanned with the scantool we "clean that up" basically. Or, to be exact, make the survival building not place those blocks.

**Comment by someaddons** [comment]
why not a setup where you auto-copy those from the original into the split parts via command blocks or so?

**Comment by Raycoms** [comment]
wdym?

**Comment by uecasm** [comment]
My point was that there should not be one schematic on the schematic server, there should be two.  One to scan for use as build tool, one to scan for use as abandoned colony.  Because these need not be the same (and in fact could usefully benefit from being different, e.g. intentional "collapse" and decay better than the random cobweb/missing block filters do), and there may be other things (e.g. paths, other connectors) that only exist in worldgen form.

Since the worldgen colonies are only level 1-2 and don't include all building types this should not take much extra space (though might be annoying to insert into existing layouts, or be located relatively far from the building they're "copying").

someaddons was suggesting that you could have a command block button to auto copy-paste from one to the other where the designer didn't want to make those sorts of custom changes for a particular building.  Though I don't think that's really necessary, the build tool can be used to do that as needed.  (Though perhaps might not be silly if the two do end up significantly distant.)  Caveat with this is that the command block would have to deal with not destroying jigsaws somehow.

**Comment by Raycoms** [comment]
Yeah, the problem is that this increases the overhead for builders, a good bit. Meanwhile, all we really need are the jigsaw blocks inside atm. Decay is done via processors anyway.
